src/ripple/app/ledger/impl/LedgerConsensusImp.cpp

```diff
@@ -1122,14 +1129,20 @@ void LedgerConsensusImp::accept (std::shared_ptr<SHAMap> set)
         LedgerMaster::ScopedLockType sl (ledgerMaster_.peekMutex (), std::defer_lock);
         std::lock(lock, sl);
 
-        // Apply transactions from the old open ledger
+        // If any transactions came in while we were building
+        // this newOL, or simply didn't survive consensus,
+        // they'll be in the old open ledger. Try to apply them
+        // now. If they're already in newLCL, they won't get
+        // reapplied. Otherwise, they'll get dropped and lost.
         Ledger::pointer oldOL = ledgerMaster_.getCurrentLedger();
         if (oldOL->txMap().getHash().isNonZero ())
         {
-            WriteLog (lsDEBUG, LedgerConsensus)
-                << "Applying transactions from current open ledger";
-            applyTransactions (&oldOL->txMap(),
-                accum, newLCL, retriableTransactions);
+            WriteLog(lsDEBUG, LedgerConsensus) <<
+                "Applying transactions from current open ledger";
+            // Apply any transactions left from the old open
+            // ledger directly to the new open ledger.
+            lowLevelApply(buildTxSet(oldOL->txMap()), accum,
```

Comment:
I have a feeling we've changed behavior here - previously it would process tx one at a time and generate an exception if it got to a bad one. With this change, we do all the deserialization up front so if there's an exception, *nothing* gets applied. This has serious consequences for transaction processing and something we don't want.